### PR TITLE
stm32h7x3: RCC definitions and patches

### PR DESCRIPTION
**Pull request by richardeoin**
* CFGR: rename fields MCOxSEL to MCOx to agree with RM
* BDCR: rename field VSWRST to BDRST to agree with RM. Rename RTCSRC to RTCSEL.
* CRRCR: rename RC48CAL to HSI48CAL
* rename RC48 to HSI48 in CR, CRRCR, CIER, CIFR and CICR
* PLL1DIVR: enumerate all even division ratios for DIVP1. The RM says
  "Note that odd division factors are not allowed." Perhaps a little
  excessive
* DxCCIPR: rename *SRC fields to *SEL to agree with RM
* APB1LRSTR: rename HDMICECRST to CECRST. APB1L(LP)ENR: rename
  HDMICECEN to CECEN.
* AHB1(LP)ENR: rename USBxOTG(LP)EN to USBxOTGHS(LP)EN. rename
  USBxULPI(LP)EN to USBxOTGHSULPI(LP)EN.
* AHB1ENR: move USB2OTGHSULPIEN to bit 18 from 28
* AHB2(LP)ENR: rename CAMITF to DCMI
* AHB3LPENR: rename FLITF to FLASH
* Optional patch to add HSICFGR/CSICFGR for Revision V
  devices. Available devices appear to be Revsion Y, so this is
  commented out

Written against RM0433 Rev 6 04-Apr-2019

### CODE REVIEW

**Comment by adamgreig** [comment]
I won't have time to review this fully until sunday/monday but as a very quick comment, patches that "fix" the SVD (i.e. use _add/_modify/_delete to update register/field names etc) should go in the `devices/` hierarchy (either in the device-specific YAML file, or broken out into `devices/common_patches/` files and included); the `peripherals/` files should more or less only contain field variants (enumerated values) or ranges (write constraints).

**Comment by richardeoin** [comment]
That comment makes perfect sense. In that case the `peripherals/` files will apply cleanly in to future SVDs that describe the same hardware. Most of the "fixes" here are specific to the h7 (for now at least), so should move to `devices/stm32h7x3.yaml`

I'll squash some commits onto this one to fix.

**Comment by jordens** [comment]
I had a look at the new SVD package for H7 (v1.4 released 2019-04-30). That has a bunch of your RCC changes implemented, many of the `_strip` actions, and also a lot of the ethernet stuff that was added from scratch. Maybe we should land that. I'll post a PR (WIP currently).

**Comment by richardeoin** [comment]
No objections to merging upstream SVD instead if the regressions against other peripherals can be cleaned up. I didn't see v1.4 and certainly wasn't going to attempt an SVD update in my first PR!

File: devices/stm32h7x3.yaml [hunk 1]
```diff
@@ -321,6 +321,97 @@ Ethernet_DMA:
...
+  BDCR:
+    _modify:
+      VSWRST:
+        name: BDRST
+      RTCSRC:
+        name: RTCSEL
+  APB1LRSTR:
+    _modify:
+      HDMICECRST:
+        name: CECRST
```
Comment (jordens) [inline devices/stm32h7x3.yaml]: This was done in the upstream SVD.

File: devices/stm32h7x3.yaml [hunk 2]
```diff
@@ -321,6 +321,97 @@ Ethernet_DMA:
...
+  D3AMR:
+    _modify:
+      BKPSRAMAMEN:
+        name: BKPRAMAMEN
```
Comment (jordens) [inline devices/stm32h7x3.yaml]: This was done in the upstream SVD.

File: devices/stm32h7x3.yaml [hunk 3]
```diff
@@ -321,6 +321,97 @@ Ethernet_DMA:
...
+  APB1LENR,C1_APB1LENR:
+    _modify:
+      HDMICECEN:
+        name: CECEN
```
Comment (jordens) [inline devices/stm32h7x3.yaml]: This was done in the upstream SVD.

File: devices/stm32h7x3.yaml [hunk 4]
```diff
@@ -321,6 +321,97 @@ Ethernet_DMA:
...
+        name: USB2OTGHSULPILPEN
+  AHB2LPENR,C1_AHB2LPENR:
+    _modify:
+      CAMITFLPEN:
+        name: DCMILPEN
+        description: "DCMI peripheral clock enable during csleep mode"
+  AHB3LPENR,C1_AHB3LPENR:
+    _modify:
+      FLITFLPEN:
+        name: FLASHPREN
```
Comment (jordens) [inline devices/stm32h7x3.yaml]: This was done in the upstream SVD.

File: devices/stm32h7x3.yaml [hunk 5]
```diff
@@ -321,6 +321,97 @@ Ethernet_DMA:
...
+      CAMITFEN:
+        name: DCMIEN
+        description: "DCMI peripheral clock"
+  APB1LLPENR,C1_APB1LLPENR:
+    _modify:
+      HDMICECLPEN:
+        name: CECLPEN
+  AHB1LPENR,C1_AHB1LPENR:
+    _modify:
+      USB1OTGLPEN:
```
Comment (jordens) [inline devices/stm32h7x3.yaml]: These were done in the upstream SVD.

File: devices/stm32h7x3.yaml [hunk 6]
```diff
@@ -321,6 +321,97 @@ Ethernet_DMA:
...
+        bitOffset: 18
+  AHB2ENR,C1_AHB2ENR:
+    _modify:
+      CAMITFEN:
+        name: DCMIEN
+        description: "DCMI peripheral clock"
+  APB1LLPENR,C1_APB1LLPENR:
+    _modify:
+      HDMICECLPEN:
+        name: CECLPEN
```
Comment (jordens) [inline devices/stm32h7x3.yaml]: This was done in the upstream SVD.

File: peripherals/rcc/rcc_v3.yaml [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        Ready: [1, "Clock ready"]
+    "*ON":
+      "Off": [0, "Clock Off"]
+      "On": [1, "Clock On"]
+  CRRCR:
+    _read:
+      HSI48CAL: [0, 0x3FF]
+  CFGR:
+    MCO2:
+      SYS: [0, "System clock selected for micro-controller clock output"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_v3.yaml]: We use `SYSCLK` everywhere else for this one.

File: peripherals/rcc/rcc_v3.yaml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      LSE: [1, "LSE selected for micro-controller clock output"]
+      HSE: [2, "HSE selected for micro-controller clock output"]
+      PLL1: [3, "PLL1 selected for micro-controller clock output"]
+      HSI48: [4, "HSI48 selected for micro-controller clock output"]
+    MCO?PRE: [0, 15]
+    SWS:
+      _read:
+        HSI: [0, "HSE oscillator used as system clock"]
+        CSI: [1, "CSI oscillator used as system clock"]
+        HSE: [2, "HSI oscillator used as system clock"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_v3.yaml]: Description should say `HSE`

**Comment by richardeoin** [comment]
Perfect, I've fixed those points. The `_strips` should have gone in the merge with #208

File: devices/stm32h7x3.yaml (hunk 6 above)
Comment (adamgreig) [reply devices/stm32h7x3.yaml]: I think this one was actually missed in both `APB1LLPENR` _and_ `C1_APB1LLPENR`

File: peripherals/rcc/rcc_v3.yaml (hunk 8 above)
Comment (richardeoin) [reply peripherals/rcc/rcc_v3.yaml]: And the HSI description should say `HSI` also

File: peripherals/rcc/rcc_v3.yaml (hunk 7 above)
Comment (richardeoin) [reply peripherals/rcc/rcc_v3.yaml]: Sure - Done

**Pull request merged**